### ISSUE TITLE
Leftover resources not being removed

### ISSUE THREAD

**Issue by nithin-nadendla** [terraform-aws-modules/terraform-aws-eks#706]
# I have issues
Terraform cannot delete leftover module resources on 'destroy'

## I'm submitting a...

* [x] bug report
* [ ] feature request
* [ ] support request - read the [FAQ](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/faq.md) first!
* [ ] kudos, thank you, warm fuzzy

## What is the current behavior?
Doing a `terraform destroy` does not delete all the module resources and gives an error.
```
Nithin-Nadendla-MBP:~/Documents/GitHub/aws-infrastructure/eks $ tf destroy
data.template_file.ecr-policy: Refreshing state...
data.terraform_remote_state.vpc: Refreshing state...
module.k8s-workflows.data.aws_caller_identity.current: Refreshing state...
module.k8s-workflows.data.aws_iam_policy_document.workers_assume_role_policy: Refreshing state...
module.k8s-workflows.data.aws_ami.eks_worker: Refreshing state...
module.k8s-workflows.data.aws_iam_policy_document.cluster_assume_role_policy: Refreshing state...
module.k8s-workflows.data.aws_ami.eks_worker_windows: Refreshing state...
module.k8s-workflows.data.aws_iam_policy_document.worker_autoscaling[0]: Refreshing state...
module.k8s-workflows.data.template_file.launch_template_worker_role_arns[0]: Refreshing state...
module.k8s-workflows.data.template_file.launch_template_worker_role_arns[1]: Refreshing state...

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:

Terraform will perform the following actions:

Plan: 0 to add, 0 to change, 0 to destroy.

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes


Error: leftover module module.trm-eks in state that should have been removed; this is a bug in Terraform and should be reported



Error: leftover module module.trm-eks.module.node_groups in state that should have been removed; this is a bug in Terraform and should be reported
```

## If this is a bug, how to reproduce? Please include a code sample if relevant.
Deploy resources using the module and then `terraform destroy`

## What's the expected behavior?
Terraform deletes all resources.

## Are you able to fix this problem and submit a PR? Link here if you have already.
No

## Environment details

* Affected module version: `v8.x.x`
* OS: `macOS Catalina 10.15.2`
* Terraform version: `Terraform v0.12.19`

## Any other relevant info
[destroy-output-debug.txt](https://github.com/terraform-aws-modules/terraform-aws-eks/files/4100040/destroy-output-debug.txt)

1. provider.aws v2.45.0
1. provider.kubernetes v1.10.0
1. provider.local v1.4.0
1. provider.null v2.1.2
1. provider.random v2.2.1
1. provider.template v2.1.2

**Comment by max-rocket-internet** [comment]
Please post the complete output showing the problem 🙂

**Comment by jayolmos** [comment]
Related: [TF - issue 22579](https://github.com/hashicorp/terraform/issues/22579) . Solution: upgrade client to terraform 0.12.20. We had the same issue and it's gone with 0.12.20.

**Comment by nithin-nadendla** [comment]
Resolved by 0.12.20
